src/jvmMain/kotlin/io/beatmaps/controllers/map.kt

```diff
@@ -118,16 +119,33 @@ fun Route.mapController() {
             headerTemplate = {
                 try {
                     transaction {
-                        Beatmap.joinVersions().select {
-                            Beatmap.id eq it.key.toInt(16) and Beatmap.deletedAt.isNull()
-                        }.limit(1).complexToBeatmap().map { MapDetail.from(it, cdnPrefix()) }.firstOrNull()
+                        Beatmap
+                            .joinCollaborators()
+                            .joinVersions()
+                            .select {
+                                Beatmap.id eq it.key.toInt(16) and Beatmap.deletedAt.isNull()
+                            }.limit(1).complexToBeatmap().map { MapDetail.from(it, cdnPrefix()) }.firstOrNull()
                     }?.let {
                         meta("og:type", "website")
                         meta("og:site_name", "BeatSaver")
                         meta("og:title", it.name)
                         meta("og:url", "${Config.siteBase()}/maps/${it.id}")
                         meta("og:image", it.publishedVersion()?.coverURL)
                         meta("og:description", it.description.take(400))
+
+                        // Joining mappers together for the og:author field so that:
+                        // 1. Uploader will be first
+                        // 2. All non-last collaborators will be joined with ", "
+                        // 3. The last collaborator will be joined with " and "
+                        val authors = (listOf(it.uploader) + it.collaborators.orEmpty()).map { u -> u.name }
+                        val authorString = authors.reduceIndexed { index, acc, s -> "$acc${if (index == authors.lastIndex) " and" else ","} $s" }
+                        meta("og:author", authorString)
+
+                        // There can only be one URL so we only add it when there is no collaborator
+                        // Otherwise, it may be confusing
+                        if (!it.collaborators.isNullOrEmpty()) {
```

Comment:
I think this is the wrong way round. The `!`, that was for the other if

Reply:
Yeah, just saw that as well. Already pushed. My bad!